### PR TITLE
UX-623 Convert EmptyState to typescript

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Converts EmptyState to typescript

### How To Test or Verify
- Verify the empty state stories work correctly

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/EmptyState/Image.tsx [hunk 1]
```diff
@@ -19,21 +26,15 @@ const Image = React.forwardRef(function Image(props, userRef) {
       height="auto"
       ref={userRef}
     >
+      {/* @ts-ignore */}
       <Picture className={className}>
         {children}
+        {/* @ts-ignore */}
```
Comment (jonambas) [inline packages/matchbox/src/components/EmptyState/Image.tsx:32]: Need to clean these up after Picture is converted

File: packages/matchbox/src/components/EmptyState/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Text, TextProps } from '../Text';
+import { Headings } from '../../helpers/types';
+
+type EmptyStateHeaderProps = {
```
Comment (logansparlin) [inline packages/matchbox/src/components/EmptyState/Header.tsx:5]: Could probably just Pick all of these from TextProps

File: packages/matchbox/src/components/EmptyState/List.tsx [hunk 3]
```diff
@@ -12,21 +11,28 @@ const StyledSvg = styled.svg`
...
       {React.Children.map(children, (child, index) => {
+        if (!React.isValidElement(child)) {
+          return null;
+        }
         return React.cloneElement(child, {
           children: (
             <div>
               {child.props.children}
-              {index < children.length - 1 && (
+              {index < React.Children.toArray(children).length - 1 && (
```
Comment (logansparlin) [inline packages/matchbox/src/components/EmptyState/List.tsx:31]: Nice

**Review by logansparlin — APPROVED** [review]
This looks good, just one small comment

**Pull request merged**